src/app/chat/view.nim

```diff
@@ -546,7 +546,11 @@ QtObject:
 
 
   proc markMessageAsSent*(self:ChatsView, chat: string, messageId: string) =
-    self.messageList[chat].markMessageAsSent(messageId)
+    if self.messageList.contains(chat):
```

Comment:
Entertained a variant of this using exceptions
```
try:
  let chatEl = self.messageList[chat]
  chatEl.markMessageAsSent(messageId)
except KeyError:
  error "Message could not be marked as sent", chat, messageId
```
but then it turns out (https://forum.nim-lang.org/t/4022) that Nim exceptions are quite expensive, even in case of error-free execution (if i read that correctly).

Reply:
Very interesting read and good to know!